### DISCUSSION TITLE
"Kata I have not trained on" filter not working

### DISCUSSION THREAD

**Discussion by mlawicki** [codewars/codewars.com#2715] [Q&A]
Hi,

Unfortunetly the possibility to filter KATAs by this filter stopped working around a week ago. I have checked on two machines and the problem is the same - it shows also KATAs I attempted to complete some time ago. I was searching and googling and did not find similar problem anywhere.

Is this an intentional behavior? Or is there another way to sort KATAs this way since last one-two weeks? ?

**Comment by kazk** [comment]
The progress filter was changed recently, and now works based on the progress in the selected language when you have a language selected.

If that's not the case, can you post the URL?

**Comment by mlawicki** [comment]
It occurs while completing katas for C#:

https://www.codewars.com/kata/search/csharp?q=&xids=played

This is the link and I see also katas that I was attempting in the past.

**Reply by kazk** [reply]
> I see also katas that I was attempting in the past.

The search shows kata you haven't trained in C#. Maybe you did in other languages?

For example, I see kata I've completed in other languages with this search:
![image](https://user-images.githubusercontent.com/639336/172482854-2e31c438-1cdd-4111-8b2a-dcabc514118c.png)

**Comment by mateuszmacheta** [comment]
I do not like current behavoir. I understand that this is cool for people who want to complete katas with more than one language. But if I'm looking for completely bew katas that I haven't even tried than this is no longer viable.

**Reply by kazk** [reply]
Then https://www.codewars.com/kata/search/my-languages?q=&xids=played should work for you, assuming you only have one language selected.

**Reply by mateuszmacheta** [reply]
I have clicked on your link @kazk, selected Python...
![IMG_20220613_090255](https://user-images.githubusercontent.com/50518772/173298067-4100c643-4b99-44c0-9265-9852892aa0b3.jpg)

...and this is what I'm getting (kata that not only I have trained on, but completed in C#)
![IMG_20220613_090337](https://user-images.githubusercontent.com/50518772/173298181-516b1ad8-30d4-451f-9d17-053def584cc0.jpg)

**Reply by kazk** [reply]
> selected Python...

Don't select the language on the search filter. That will do language-specific search that most users expect.

To get what you want, select one language in https://www.codewars.com/trainer/setup and use "My Languages" filter like the link I posted.

**Reply by mateuszmacheta** [reply]
Thank you @kazk, that'll do.
